gubernator/gcs_async.py

```diff
@@ -36,7 +36,12 @@ def get(url):
         if status in (200, 206):
             content = result.content
             if result.headers.get('content-encoding') == 'gzip':
-                content = zlib.decompress(result.content, 15 | 16)
+                dec = zlib.decompressobj(15 | 16)
+                content = dec.decompress(result.content, MAX_SIZE)
+                if dec.unconsumed_tail:
+                    logging.warning('only decompressed %d KB, %d KB remain in buffer.',
```

Comment:
how does the user see this?

Reply:
They see a partial (truncated) log file.

They always see a partial log file, since the highlighter picks up likely interesting keywords, so I don't think the fuzziness matters much.

These pathological log files apparently include core dumps in hex format (???).

Reply:
Can we surface the message somehow in the UI at all? Some of our very verbose logs get long... but we truncate them before we upload them, the first line we replace with a banner explaining that they were truncated. Something that would make the user less likely to open a bug against Gubernator in the future where they complain that Gubernator is acting weird and showing only half of a line :)